### ISSUE TITLE
Bug when related entries link to each other

### ISSUE THREAD

**Issue by jjdb** [josephwright/biblatex-phys#17]
Dear Joseph Wright,

I'm a happy user of your biblatex-phys package. However, I believe, I've encountered a bug in the package. 

The main file is test.tex and the included bibliography Bibliography.bib. 

I've used a similar structure in my thesis, for which I used TexLive2014. I've tried to use the related entries feature of biblatex, but I believe that it was not supported at that time or for the version of biblatex I've used. The files compiled without problem then (but as I said, the related entries did not show up. I remember I've wondered why, but now I believe it could be because I've used an older version of biblatex but tried features from a more current version of the documentation). 

Then recently, I've upgraded to TexLive2016, with the document not compiling any more in the run after biber when including the command \printbibliography. The compilation process simply stops when reaching the bibliography, and exits after some minutes with an useless error message. I've tracked the issue now down to two references which referred to each other by the related label and with the same related-type (stupid, I know). The document compiles fine if I remove either \printbibliography, the related information of one of the entries, or if I choose the standard bibstylefile instead (i.e. removing the line bibstyle=phys,). So without the package, it compiles fine.

In any case, the problem should not occur to often (as not everybody is so stupid as I am), however, it took me quite some time to track down the issue as neither meaningful error message is showing up after the second compilation nor a single compilation of any source files shows an error message at all.

Nevertheless, thank you for all the time and effort put into this package. I hope you find this message helpful. In any case you can contact me for further information if needed.

Best regards,
jjdb

PS: Please apologize if anything in this message sounds rude, I'm not a native English speaker.

test.tex

```
\documentclass[ %
                twoside,
                openright,
                footinclude=true,
                cleardoublepage=current,
                paper=a4,
                Fontsize=11pt,
                ]{scrreprt}
\usepackage[utf8]{inputenc}
\usepackage[    
      backend=biber,
      bibstyle=phys,
      ]{biblatex}
\addbibresource{./Bibliography.bib}
        \usepackage{hyperref}  
\begin{document}
\chapter{Chapter 1}
a
\chapter{Chapter 2}
b

This is a citation of \parencite{ref1}
\printbibliography
\end{document}
```

Bibliography.bib

```
@article{ref1,
  author={author1},
  journal={journal1},
  title = {{title1}},
  year={2001},
  month={March},
  volume={1},
  number={2},
  pages={83-87},
  doi={10.1234/876543219},
  related = {ref2},
  relatedtype = {reprintof},
}
@article{ref2,
  author={author2},
  journal={journal2},
  title = {{title2}},
  year={2002},
  month={June},
  volume={2},
  number={2},
  pages={353-789}, 
  doi={10.1234/987654321},
  related = {ref1},
  relatedtype = {reprintof},
}
```

**Comment by josephwright** [comment]
This is not a `biblatex-phys` issue _per se_, but it would show up with any style. However, detecting circular references can be tricky and I think this is likely just 'one of those things'.

**Comment by jjdb** [comment]
No, I think I put it confusingly above. It compiles fine if I use 'bibstyle=alphabetic' e.g. So I suspect there must be an issue here.

**Comment by jjdb** [comment]
I did just again make some more testing after checking the source of `biblatex-phys`. I was first thinking the `reprintof` specification is making the problem, that one entry can't be the reprint of the reprint of itself. However, you don't specify `reprintof` anywhere in your code. Then I've tried it with `translatedas` as this is specified in the code, which to my surprise compiles fine (both entries are related by the relatedtype `translatedas`). 

I then had a look at the `biblatex` doc, and I found the following 

> Secondly, if there is a macro called related:<relatedtype>, this is used to format the data from the
related entries. If no such macro exists, then the macro related:default is used.

Then I tried to compile with both having a relatedtype `qwerty` )I've made that up): doesn't compile. 
Having one entry `qwerty` and one `translatedas`: it compiles. 
Leaving away both relatedtype entries: doesn't compile. 

So I conclude that there has to be an error in the part of the code where the `related:default ` is defined. 

I hope this is of help for you.
